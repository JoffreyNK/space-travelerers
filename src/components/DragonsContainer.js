import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getDragon } from '../redux/dragons/dragonSlice';
import AllDragons from './AllDragons';

const DragonsContainer = () => {
  // Using Use Selector to grab the state
  const alldragons = useSelector((state) => state.dragon);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!alldragons.length) {
      dispatch(getDragon());
    }
  }, [alldragons.length]);

  return (
    <div className="rockets-container">
      {
      alldragons.length ? <AllDragons dragon={alldragons} /> : <div>No Dragon Found</div>
      }
    </div>
  );
};

export default DragonsContainer;
