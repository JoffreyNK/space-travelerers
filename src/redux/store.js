import { configureStore, combineReducers } from '@reduxjs/toolkit';
import missionsSlice from './missions/missionsReducer';

const reducer = combineReducers(
  {
    missions: missionsSlice.reducer,
  },
);

const store = configureStore({ reducer });

export default store;
