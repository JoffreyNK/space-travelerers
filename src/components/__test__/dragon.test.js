// Dragon.test.js

import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import DragonsContainer from '../DragonsContainer';
import store from '../../redux/store';

describe('Dragon Component Rendering ', () => {
  it('Matches snapshot', () => {
    const Tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <DragonsContainer />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(Tree).toMatchSnapshot();
  });
});
