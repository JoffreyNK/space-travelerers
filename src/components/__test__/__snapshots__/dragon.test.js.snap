// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragon Component Rendering  Matches snapshot 1`] = `
<div
  className="rockets-container"
>
  <div>
    No Dragon Found
  </div>
</div>
`;

exports[`Dragon Component Rendering  renders correctly 1`] = `
<div
  className="rockets-container"
>
  <div>
    No Dragon Found
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="rockets-container"
>
  <div>
    No Dragon Found
  </div>
</div>
`;
